internalv2compat/compat.go

```diff
@@ -137,3 +140,11 @@ func V2TracingHTTPServerMiddleware() func(name string, next http.Handler) http.H
 	defer v2Tracing.Unlock()
 	return v2Tracing.httpServerMiddleware
 }
+
+func SetV0BaseplateHeadersToContext(ctx context.Context, headers map[string]string) context.Context {
+	return headerbp.ToContext(ctx, headers)
+}
+
+func SetV2BaseplateHeadersLookup(setter func(context.Context, map[string]string) context.Context) {
+	headerbp.SetV2Context = setter
```

Comment:
One thing I noticed with our `internal2compat` primarily sets on structs local to this package but then makes it accessible through a separate getter. Would it make sense to define this locally here too and then call `V2Context()` to get the function outside of this package?

Reply:
I see, that makes sense, updated.